pymc/distributions/timeseries.py

```diff
@@ -114,109 +116,131 @@ def get_steps(
+            steps=steps,
+            shape=kwargs.get("shape"),
+            step_shape_offset=1,
+        )
+        if steps is None:
+            raise ValueError("Must specify steps or shape parameter")
+        steps = at.as_tensor_variable(intX(steps))
 
-        Returns
-        -------
-        ndarray
-        """
+        if not (
+            isinstance(init_dist, at.TensorVariable)
+            and init_dist.owner is not None
+            and isinstance(init_dist.owner.op, (RandomVariable, SymbolicRandomVariable))
+            # TODO: Lift univariate constraint on init_dist
+            and init_dist.owner.op.ndim_supp == 0
+        ):
+            raise TypeError("init_dist must be a univariate distribution variable")
+        check_dist_not_registered(init_dist)
+
+        if not (
+            isinstance(innovation_dist, at.TensorVariable)
+            and init_dist.owner is not None
+            and isinstance(init_dist.owner.op, (RandomVariable, SymbolicRandomVariable))
+            # TODO: Lift univariate constraint on inovvation_dist
+            and init_dist.owner.op.ndim_supp == 0
+        ):
+            raise TypeError("init_dist must be a univariate distribution variable")
+        check_dist_not_registered(init_dist)
+
+        return super().dist([init_dist, innovation_dist, steps], **kwargs)
 
-        if steps < 1:
-            raise ValueError("Steps must be greater than 0")
+    @classmethod
+    def rv_op(cls, init_dist, innovation_dist, steps, size=None):
+        if not steps.ndim == 0 or not steps.dtype.startswith("int"):
+            raise ValueError("steps must be an integer scalar (ndim=0).")
 
-        # If size is None then the returned series should be (*implied_dims, 1+steps)
+        # If not explicit, size is determined by the shapes of the input distributions
         if size is None:
-            # broadcast parameters with each other to find implied dims
-            bcast_shape = np.broadcast_shapes(
-                np.asarray(mu).shape,
-                np.asarray(sigma).shape,
-                np.asarray(init_dist).shape,
-            )
-            dist_shape = (*bcast_shape, int(steps))
-
-        # If size is None then the returned series should be (*size, 1+steps)
-        else:
-            dist_shape = (*size, int(steps))
-
-        # Add one dimension to the right, so that mu and sigma broadcast safely along
-        # the steps dimension
-        innovations = rng.normal(loc=mu[..., None], scale=sigma[..., None], size=dist_shape)
-        grw = np.concatenate([init_dist[..., None], innovations], axis=-1)
-        return np.cumsum(grw, axis=-1)
-
-
-gaussianrandomwalk = GaussianRandomWalkRV()
+            size = at.broadcast_shape(init_dist, innovation_dist[..., 0])
+        innovation_size = tuple(size) + (steps,)
+
+        # Resize input distributions
+        init_dist = change_dist_size(init_dist, size)
+        innovation_dist = change_dist_size(innovation_dist, innovation_size)
```

Comment:
If this line anticipated `innovation_dist` that's already a vector, one could do random walks that have different innovations in every step, right? (e.g. via `at.stack([Normal.dist(), Uniform.dist(), Beta.dist()])`

Reply:
This requires more discussion: https://github.com/pymc-devs/pymc/discussions/6090